### PR TITLE
Improve Huawei VRP bgp discovery

### PR DESCRIPTION
**Pull request by PipoCanaja**
This PR is improving multi-VRF BGP discovery on Huawei CloudEngine and AR devices. Polling is not touched for the moment (so BGP data is still missing during polling) because a more important change (DB) is probably needed here (that will necessarily impact other OSes).

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 11054`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: includes/discovery/functions.inc.php [hunk 1]
```diff
@@ -1248,10 +1248,10 @@ function build_cbgp_peers($device, $peer, $af_data, $peer2)
 
         $afisafi_tmp = explode('.', $k);
         if ($device['os_group'] === 'vrp') {
-            array_shift($afisafi_tmp); //remove 1st value, always 0 so far
+            $vpninst_id  = array_shift($afisafi_tmp);
```
Comment (PipoCanaja) [inline includes/discovery/functions.inc.php]: This variable is not used (yet) but will be as soon as this vpninst_id is stored properly.

File: includes/discovery/functions.inc.php [hunk 2]
```diff
@@ -1248,10 +1248,10 @@ function build_cbgp_peers($device, $peer, $af_data, $peer2)
...
             $afi         = array_shift($afisafi_tmp);
             $safi        = array_shift($afisafi_tmp);
-            array_shift($afisafi_tmp); //type, always ipv4 so far
+            $peertype    = array_shift($afisafi_tmp);
```
Comment (PipoCanaja) [inline includes/discovery/functions.inc.php]: Same here, peer type (ipv4 vs ipv6) is not stored for now.

**Comment by PipoCanaja** [comment]
Hi Guys,
Ready for review. I would like to merge this one, and then I'll open a new one to split BGP peer discovery for each OS (and keep the default BGP4-MIB as last one if the OS-Specific does not return anything ). That would make the discovery file for BGP simpler to read / troubleshoot / extend. 
Thanx

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/bgp-sessions-under-vrf/5717/8

**Comment by Jellyfrog** [comment]
@PipoCanaja couldn't you generate test data anyway?

**Comment by PipoCanaja** [comment]
Yep, as soon as I have this use case in my lab. Right now I only have it in prod, and would need to forge the SNMPREC file manually ... doable but boring :)

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/huawei-ne40e-mk2b-peer-bgp-disappeared-after-system-update/10681/5

**Comment by PipoCanaja** [comment]
tests on their way

**Comment by PipoCanaja** [comment]
Tests done. This time, it looks ready for review and merge.

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-61-release-changelog-february-2020/11145/1

**Pull request merged**